core/src/main/java/io/questdb/std/str/AbstractCharSequence.java

```diff
@@ -57,6 +57,14 @@ public final CharSequence subSequence(int start, int end) {
         if (start == 0 && end == length()) {
             return this;
         }
+        // it's tempting apply this optimization:
+        // if (start == end) {
+        //   return "";
+        // }
+        // but that's rather risky, because for example GenericLexer does this:
+        // if (!(value0 instanceof FloatingSequence && value1 instanceof InternalFloatingSequence)) {
```

Comment:
this piece of code is a good spot, perhaps it should be a `assert`. In light of recently discovered JIT bug with cache line thrashing. Or perhaps even removed altogether. ClassCastException is good enough

Reply:
agree, removing it should not make anything worse for users.